scripts/get_charm_states.py

```diff
@@ -0,0 +1,70 @@
+"""Script to download from a tempo backend the charm traces associated with a specific unit."""
+import json
+from pathlib import Path
+from typing import Optional
+
+import requests
+import typer
+
+app = typer.Typer()
+
+
+def _get_charm_states(target: str, tempo: str, output: Optional[Path]=None):
+    # get trace IDs for this target
+    target_name = target.replace("/", "-")
+    req = requests.get(
+        f"{tempo}/api/search",
+        params={
+            "q": f'{{rootServiceName = "{target_name}"}}'
```

Comment:
If you agree with not changing the service name to contain the unit (comment in `charm_tracing` above)  you may need to do the split here to search for `rootServiceName` and `unitName`.

Reply:
ah, we already have j.uju_unit from the juju topology, we can do that either way. Fixed